src/include/ipc/dai.h

```diff
@@ -61,29 +62,53 @@ enum sof_ipc_dai_type {
 	SOF_DAI_INTEL_ALH,		/**< Intel ALH */
 	SOF_DAI_IMX_SAI,                /**< i.MX SAI */
 	SOF_DAI_IMX_ESAI,               /**< i.MX ESAI */
+	SOF_DAI_MULTIDAI,		/**< MultiDAI */
 };
 
-/* general purpose DAI configuration */
-struct sof_ipc_dai_config {
-	struct sof_ipc_cmd_hdr hdr;
+struct sof_ipc_dai_info {
 	uint32_t type;		/**< DAI type - enum sof_ipc_dai_type */
 	uint32_t dai_index;	/**< index of this type dai */
+} __attribute__((packed));
 
+struct sof_ipc_dai_base_params {
 	/* physical protocol and clocking */
 	uint16_t format;	/**< SOF_DAI_FMT_ */
 	uint16_t reserved16;	/**< alignment */
 
 	/* reserved for future use */
-	uint32_t reserved[8];
+	uint32_t reserved[4];
+} __attribute__((packed));
+
+struct sof_ipc_multidai_params {
+	struct sof_ipc_cmd_hdr hdr;
+
+	uint32_t num_dais;
+	uint32_t reserved[5];
+
+	struct sof_ipc_stream_map stream_map;
+	struct sof_ipc_dai_info info[0];
+	struct sof_ipc_dai_base_params params[0];
```

Comment:
This is difficult to index, best to do 
```
struct ipc_dai_blah {
    struct sof_ipc_dai_info info;
    struct sof_ipc_dai_base_params params;
};
```
then
```
struct sof_ipc_multidai_params {
...
    struct ipc_dai_blah dai_blah[0];
};
```
This way info and params can be addressed using regular array indexing (instead of manual calculations).